src/App.tsx

```diff
@@ -1,24 +1,198 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import {
+  useCallback,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { Header } from './components/header';
+import { Main } from './components/main';
+import { Footer } from './components/footer';
+import { Notification } from './components/notification';
+import { Todo } from './types/Todo';
+import { USER_ID } from './utils/constants';
+import { Filter } from './types/Filter';
+import {
+  getTodos,
+  deleteTodo,
+  updateTodoCompleted,
+  updateTodoTitle,
+} from './api/todos';
 
-const USER_ID = 0;
+export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState<string>('');
+  const [filter, setFilter] = useState<string>(Filter.ALL);
+  const [loadingID, setLoadingID] = useState<number[]>([]);
+  const [comletedTodos, setCompletedTodos] = useState<Todo[]>([]);
+  const [uncomletedTodoCount, setUncomletedTodoCount] = useState<number>(0);
 
-export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const visibleTodos = useMemo(() => {
+    let filteredTodos: Todo[] = todos;
+
+    if (filter === Filter.ACTIVE) {
+      filteredTodos = todos.filter(todo => !todo.completed);
+    }
+
+    if (filter === Filter.COMPLETED) {
+      filteredTodos = todos.filter(todo => todo.completed);
+    }
+
+    return filteredTodos || [];
```

Comment:
Filter always return an array
```suggestion
    return filteredTodo;
```